### ISSUE TITLE
Naming conventions/rules for all checkpoints

### ISSUE THREAD

**Issue by kargibora** [KarahanS/Self-Supervised-Learning-for-Medical-Image-Analysis#9]
resnet50_simclr_200_256_pt=True_s=42
resnet50_simclr_800_256_pt=True_s=42
downstream-linear-eval_resnet50_pathmnist

I don't think the naming scheme provided above is clear enough - we have to find a proper naming convention so that we can keep track of our checkpoints

**Comment by kargibora** [comment]
# Ideas
We shouldnt use any boolean values or seed informations in the config names. 

`{backbone}_{ssl_method}_{optimizer}_{lr}_{epochs}_{batch}_(specific informations)` will be enough. Where we can provide specific informations in the Config that @kivanctezoren  will implement.

**Comment by KarahanS** [comment]
Are we planning to change/compare optimizers or learning rates for a specific SSL method? Otherwise, it seems redundant to keep `optimizer` and `lr` information in the name. 
Additionally, I think @kivanctezoren will adjust the config so that we can provide different augmentations and use different pre-trained weights. So we should be storing them as well.
For reproducibility and to differentiate the runs with different seeds (rest is the same), we should store the seed information.

**Comment by OguzAtaCal** [comment]
We might compare/change optimizers or learning rates also for other SSL methods? Because we did some testing with SimCLR due to implementation differences between ours and the papers.
The `specific information` field at the end like @kargibora suggested seems like a good idea to me. We can also remove the `lr` and `optimizer ` field from the naming convention and add them to the `specific information` field as we do specific tests.
